EATSSU/App/Sources/Data/Firebase/LaunchSourceManager.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  LaunchSourceManager.swift
+//  EATSSU
+//
+//  Created by 황상환 on 3/23/25.
+//
+
+import Foundation
+import UIKit
+import FirebaseAnalytics
+
+enum LaunchSource: String {
+    case icon = "icon"
+    case localNotification = "local_notification"
+    case widget = "widget"
+    case background = "background"
+    
+}
+
+class LaunchSourceManager {
+    static let shared = LaunchSourceManager()
+    
+    private(set) var source: LaunchSource = .icon
+    private var hasLogged = false
+    private var backgroundEntryTime: Date?
+    private let newSessionThreshold: TimeInterval = 300
+    
+    private init() {}
+    
+    /// 앱 실행 소스를 설정합니다
+    /// - Parameter source: 설정할 실행 소스
+    func setSource(_ source: LaunchSource) {
+        print("소스 설정 시도: \(source.rawValue) (현재 소스: \(self.source.rawValue), 로깅 상태: \(hasLogged))")
+        
+        // 로컬 알림 소스 설정은 항상 우선시
+        if source == .localNotification {
+            let changed = self.source != source
+            self.source = source
+            
+            // 소스가 변경되었으면 로깅 상태 초기화
+            if changed {
+                hasLogged = false
+            }
+            return
+        }
+        
+        // 위젯 소스 설정도 항상 우선시
+        if source == .widget {
+            let changed = self.source != source
+            self.source = source
+            
+            if changed {
+                hasLogged = false
+            }
+            return
+        }
+        
+        // 아이콘 소스는 기존 소스가 없을 때만 설정
+        if source == .icon && self.source == .icon {
+            // 이미 아이콘으로 설정되어 있으면 변경 없음
+            return
+        }
```

Comment:
switch문으로 바꾸면 가독성이 더 좋아질 것 같네요

```suggestion
        // 로컬 알림 소스 설정은 항상 우선시
        
        switch source {
            case .localNotification, .widget:
               let changed = self.source != source
               self.source = source
               if changed {
                   hasLogged = false
               }
               return
           case .icon:
              if self.source == .icon {
                   return // 이미 icon이면 아무것도 하지 않음
              }
             // default처럼 처리됨
             fallthrough
           default:
                 break 
    }
}
```